Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -409,8 +412,8 @@ namespace pcpp
 			return;
 		}
 
-		bool sameDescriptor = (m_PcapDescriptor == m_PcapSendDescriptor);
-		pcap_close(m_PcapDescriptor);
+		bool sameDescriptor = (m_PcapDescriptor.get() == m_PcapSendDescriptor);
+		m_PcapDescriptor = nullptr;
```

Comment:
We probably don't need this line because it will be destroyed automatically?

Reply:
`close` can be called in other places instead of only in the destructor, therefor we can't assume that `m_pcapDescriptor` will be destroyed and rely on that.

Reply:
Not a fan for setting `nullptr` as the way to close the handle. Maybe we should make the close function public and call `m_PcapDescriptor.close()`?

Reply:
Sure, we can expose a public function to close the handle.

Reply:
Added a function `reset()` mirroring `std::unique_ptr`'s implementation, that can be used to close the handle.